Sources/SwiftKafka/KafkaConsumer.swift

```diff
@@ -371,11 +423,51 @@ extension KafkaConsumer {
                     source: source
                 )
                 return .setUpConnection(client: client)
-            case .consuming, .finishing, .finished:
+            case .consuming, .consumptionStopped, .finishing, .finished:
                 fatalError("\(#function) should only be invoked upon initialization of KafkaConsumer")
             }
         }
 
+        /// The messages asynchronous sequence was terminated.
+        /// All incoming messages will be dropped.
+        mutating func stopConsuming() {
+            switch self.state {
+            case .uninitialized:
+                fatalError("\(#function) invoked while still in state \(self.state)")
+            case .initializing:
+                fatalError("Call to \(#function) before setUpConnection() was invoked")
+            case .consumptionStopped:
+                fatalError("stopConsuming() must not be invoked more than once")
+            case .consuming(let client, let source):
+                self.state = .consumptionStopped(client: client, source: source)
+            case .finishing, .finished:
+                break
+            }
+        }
+
+        /// Action to take when wanting to store a message offset (to be auto-committed by `librdkafka`).
+        enum StoreOffsetAction {
+            /// Store the message offset with the given `client`.
+            /// - Parameter client: Client used for handling the connection to the Kafka cluster.
+            case storeOffset(client: KafkaClient)
+        }
+
+        /// Get action to take when wanting to store a message offset (to be auto-committed by `librdkafka`).
+        func storeOffset() -> StoreOffsetAction {
```

Comment:
Can we call this `messageSequenceNext` and also rename the action. It is better to name those methods how the places are called and not with the intended outcome.

Reply:
I disagree on this one 😅 -- why do you think it is better?

I think it's better if the action you perform on the state machine is named after the thing you want to do as it makes the call site easier to understand. If `messageSequenceNext()` calls `state.messageSequenceNext()` I have no idea what's actually happening. If it calls `state.storeOffet()` I have _some_ idea of what's happening.

A point against naming it against the call site is that it's also just unnecessary coupling.

Reply:
I am a bit torn on this one. I do think naming it like the call site is beneficial because in most of these state machines the action we return depends on what we do where. So it is already very coupled.